### PR TITLE
Metadata manager part 1

### PR DESCRIPTION
**Pull request by zirrock**

### CODE REVIEW

**Review by avelanarius — APPROVED** [review]
LGTM, if you want you can make those small fixes.

File: src/kafka/producer/metadata_manager.hh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+private:
+    lw_shared_ptr<connection_manager> _connection_manager;
+    metadata_response _metadata;
+
+public:
+    metadata_manager(lw_shared_ptr<connection_manager> &manager)
+    : _connection_manager(manager) {}
+
+    seastar::future<metadata_response> refresh_metadata();
+    metadata_response &get_metadata();
```
Comment (constantfold) [inline src/kafka/producer/metadata_manager.hh]:
```suggestion
    metadata_response& get_metadata();
```

File: src/kafka/producer/metadata_manager.hh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace kafka {
+
+class metadata_manager {
+
+private:
+    lw_shared_ptr<connection_manager> _connection_manager;
+    metadata_response _metadata;
+
+public:
+    metadata_manager(lw_shared_ptr<connection_manager> &manager)
```
Comment (constantfold) [inline src/kafka/producer/metadata_manager.hh]:
```suggestion
    metadata_manager(lw_shared_ptr<connection_manager>& manager)
```

File: src/kafka/producer/kafka_producer.cc [hunk 3]
```diff
@@ -45,106 +45,93 @@ namespace seastar {
...
-
-seastar::future<metadata_response> kafka_producer::refresh_metadata() {
-    kafka::metadata_request req;
-
-    req._allow_auto_topic_creation = true;
-    req._include_cluster_authorized_operations = true;
-    req._include_topic_authorized_operations = true;
-
-    return _connection_manager.ask_for_metadata(req);
+    metadata_response &metadata = _metadata_manager.get_metadata();
```
Comment (constantfold) [inline src/kafka/producer/kafka_producer.cc]:
```suggestion
    metadata_response& metadata = _metadata_manager.get_metadata();
```

File: src/kafka/producer/metadata_manager.cc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        req._allow_auto_topic_creation = true;
+        req._include_cluster_authorized_operations = true;
+        req._include_topic_authorized_operations = true;
+
+        return _connection_manager->ask_for_metadata(req).then([this] (metadata_response metadata){
+            return (_metadata = metadata);
+        });
+    }
+
+    metadata_response &metadata_manager::get_metadata() {
```
Comment (constantfold) [inline src/kafka/producer/metadata_manager.cc]:
```suggestion
    metadata_response& metadata_manager::get_metadata() {
```

**Pull request merged**